src/components/cc-invoice/cc-invoice.types.d.ts

```diff
@@ -0,0 +1,20 @@
+export type InvoiceState = InvoiceStateLoading | InvoiceStateError | InvoiceStateLoaded;
+
+interface InvoiceStateLoading {
+  type: 'loading';
+  number?: string;
+}
+
+interface InvoiceStateError {
+  type: 'error';
+  number: string;
```

Comment:
question: is there a reason to not be optional here while it is on `loading`?

Reply:
Yes, the error comes from remote call which needs the number. So, there is always a number.